src/emqtt.erl

```diff
@@ -736,6 +747,10 @@ init([{bridge_mode, Mode} | Opts], State) when is_boolean(Mode) ->
     init(Opts, State#state{bridge_mode = Mode});
 init([{reconnect, IsReconnect} | Opts], State) when is_boolean(IsReconnect) ->
     init(Opts, State#state{reconnect = IsReconnect});
+init([{reconnect, Attempts} | Opts], State) when is_integer(Attempts) ->
+    init(Opts, State#state{reconnect = Attempts});
```

Comment:
The `infinity` variant doesn't seem to be handled. Nit: it's now unclear to a user what's the difference between `{reconnect, true}` and `{reconnect, infinity}`. Nevertheless, I'd also propose to normalize all those variants to `non_neg_integer() | infinity` here (especially given that `num_of_reconnect_attempts/1` does not handle `false`).

Reply:
The main reason is that our configuration options are preferably compatible with older versions.
Or we can use the new definition in the type declaration but be compatible with boolean() when parsing the user's option?

Reply:
Compatibility is important concern, no questions here.

> Or we can use the new definition in the type declaration but be compatible with boolean() when parsing the user's option?

My only concern is that the dialyzer will probably complain then. Let's leave this option as it is, albeit a bit redundant.